metatensor-torch/src/labels.cpp

```diff
@@ -13,10 +13,15 @@ using namespace metatensor_torch;
 /// Check that `values` is a `shape_length`-dimensional array of 32-bit
 /// integers, or convert it to be so.
 static torch::Tensor normalize_int32_tensor(torch::Tensor values, size_t shape_length, const std::string& context) {
+    // if the tensor is empty, make sure the shape is (0, len(sample_names))
+    if (!values.numel()) {
+        values = torch::empty(values.sizes(), torch::TensorOptions().dtype(torch::kInt32));
```

Comment:
```suggestion
        values = torch::empty(values.sizes(), torch::TensorOptions().dtype(torch::kInt32).device(values.device()));
```

we also need the same device